src/rules/jsonFlag.ts

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { ESLintUtils } from '@typescript-eslint/utils';
+import { ancestorsContainsSfCommand, isInCommandDirectory } from '../shared/commands';
+import { flagPropertyIsNamed, isFlag } from '../shared/flags';
+
+export const jsonFlag = ESLintUtils.RuleCreator.withoutDocs({
+  meta: {
+    docs: {
+      description: 'do not allow creation of json flag',
+      recommended: 'error',
+    },
+    messages: {
+      message: 'It is not necessary to add a --json flag.  That flag is provided by sfCommand/oclif',
```

Comment:
Is it possible to provide a link to how `sfCommand/oclif`'s `--json` flag works so they know what they get for free? 

Maybe there isn't a super good place to send them, though. On sf-plugin-core, I only see mention of ["Enable the json flag support by default"](https://github.com/salesforcecli/sf-plugins-core#sfcommand-abstract-class). https://oclif.io/docs/json is OK, but someone might interpret that they need to add `enableJsonFlag` to their command. In order to do things right, they need to combine the knowledge of both oclif.io and that passing `sfCommand` comment to add the type. Not super clear for a newbie.